### PR TITLE
Refactor min assurance for resolution

### PR DESCRIPTION
**Pull request by dlongley**

### CODE REVIEW

File: lib/entities.js [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+ * Increments an entity's `batchInvalidationCount` to indicate that all
+ * unpinned batches are about to be invalidated -- and any that are
+ * concurrently created should be considered invalid. The update will only
+ * be applied if the entity exists and its `batchInvalidationCount` has not
+ * changed from the given value. This is to ensure that any decision made to
+ * invalidate token batches based on a previously read value will be consistent
+ * with the current state.
+ *
+ * @param {object} options - Options to use.
+ * @param {string} options.entity - The entity.
```
Comment (mandyvenables) [inline lib/entities.js:95]: The entity should be an object here.

File: lib/entities.js [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    query, {$set}, database.writeOptions);
+
+  // return `true` if the update occurred
+  return result.result.n !== 0;
+}
+
+/**
+ * Upserts a new entity. If the entity already exists, its time to live will be
+ * updated, but its `minAssuranceForResolution` will not be changed. To change
+ * the entity's `minAssuranceForResolution`
```
Comment (mandyvenables) [inline lib/entities.js:174]: It looks like this comment is unfinished.

Comment (dlongley) [reply lib/entities.js]: Thanks! Fixed in c4336da90e23b8c91bd395a755a3bfa42299b1d3.

File: lib/entities.js (hunk 1 above)
Comment (dlongley) [reply lib/entities.js]: Thanks! Fixed in c4336da90e23b8c91bd395a755a3bfa42299b1d3.

**Review by mandyvenables — APPROVED** [review]
Looks good, and passes tests I was previously using.

**Review by dmitrizagidulin — APPROVED** [review]
:+1: 
Thanks @dlongley !

**Comment by dmitrizagidulin** [comment]
Merging into the `add-minAssuranceForResolution` branch/PR.

**Pull request merged**